### PR TITLE
Rapid Antigen Tests: Reject manipulated non-personalized QR codes

### PR DESCRIPTION
**Pull request by freshking**
## Description

Add validation for non-personalized QR codes.

## Link to Jira

https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6925

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift [hunk 1]
```diff
@@ -47,33 +47,36 @@ enum Route {
...
-				let recomputedHashString = ENAHasher.sha256(informationString)
-				guard recomputedHashString == testInformation.hash else {
-					self = .rapidAntigen( .failure(.invalidTestCode(.hashMismatch)))
-					Log.error("recomputed hash: \(recomputedHashString) Doesn't match the original hash \(testInformation.hash)", log: .qrCode)
-					return
-				}
+				recomputedHashString = ENAHasher.sha256(informationString)
 			}
+			
+			guard recomputedHashString == testInformation.hash else {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift:74]: Can we have a test for that?

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift [hunk 2]
```diff
@@ -47,33 +47,36 @@ enum Route {
...
-			}
+			let firstName = testInformation.firstName ?? ""
+			let lastName = testInformation.lastName ?? ""
+			let dateOfBirthString = testInformation.dateOfBirthString ?? ""
+			let salt = testInformation.cryptographicSalt ?? ""
+			let timestamp = String(testInformation.timestamp)
+			let recomputedHashString: String
 			
-			if allIsValid {
+			if firstName.isEmpty && lastName.isEmpty && dateOfBirthString.isEmpty {
```
Comment (ArturFriesen) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift:58]: From the tech spec: "dob` (date of birth) must have the format `YYYY-MM-DD`"
I'm missing it here. Do I have Tomaten auf den Augen?

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift (hunk 1 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift:74]: Working on it

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift (hunk 2 above)
Comment (freshking) [reply src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift:58]: This is not part of the ticket or this PR

File: src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift [hunk 3]
```diff
@@ -15,7 +15,7 @@ enum Route {
 		}
 		self.init(url: url)
 	}
-	// swiftlint:disable:next cyclomatic_complexity
+	// swiftlint:disable:next cyclomatic_complexity superfluous_disable_command
```
Comment (nickguendling) [inline src/xcode/ENA/ENA/Source/AppDelegate & Globals/Route.swift:18]:
```suggestion
	// swiftlint:disable:next cyclomatic_complexity
```
Seems like its still needed, so no superfluous command there anymore

**Review by KaiTeuber — APPROVED** [review]

**Review by nickguendling — CHANGES_REQUESTED** [review]

**Review by ArturFriesen — APPROVED** [review]

**Review by nickguendling — APPROVED** [review]

**Pull request merged**